### ISSUE TITLE
Too-large items should include multiple items

### ISSUE THREAD

**Issue by Noodleyman** [dvdoug/BoxPacker#188]
I think this request would be considered an enhancement request.

In reference to the example provided here:
https://www.boxpacker.io/en/stable/too-large-items.html

Based on this example and some testing, it appears that when ItemToLargeException is thrown it will only ever include a single item. (as the exception name implies). 

In some cases there may be multiple items that cannot be packed, however in this situation only a single item is returned.

It would be quite useful in some situations if a List of all items which are not able to be packed be returned. This would make handling those items via business specific criteria much more efficient.

**Comment by dvdoug** [comment]
Will think about it - the exception as thrown is supposed to be a truly exceptional case whereby something must have gone wrong. If too large items are "normal" for a flow, `InfalliblePacker` (documented on the same page) should be a better fit than the regular `Packer`?

**Comment by dvdoug** [comment]
Or, based on your other issue are you just using `VolumePacker` directly rather than the `Packer` wrapper? That would be a usecase that's perhaps not ideally handled

**Comment by Noodleyman** [comment]
Good Morning,

Thanks for your feedback. I am probably using BoxPacker in an unusual way compared to most given the problem I am solving. I have to do multiple passes / steps based on the inventory being packed. 

I have a list of boxes which are rated for specific licensed materials. my first step is to pack those licensed items first into the smallest suitable packing boxes. The items also make use of custom constraints. I do not use weight balancing for this step.

As special permit boxes are expensive (can be up to $20 per box) they are strictly inventory controlled. I implemented stock control, see #169 which is working correctly.

A consequence of running out of boxes of a specific size and license type could mean that there are no boxes remaining to pack items. For example if I have 10 items which require 5 boxes, but only have one suitable box available. In this situation I would end up with 8 items that have no suitable boxes and 8 items which cannot be packed.  It would be helpful to have the 8 items returned, then I can loop those and handle via custom logic in a single pass rather than having to rebuild the list, exclude the one item and repeat the process until everything is either packed or marked as non-packable.

When there are no suitable boxes the ItemTooLargeException is thrown. in this case that could mean we ran out of suitable boxes or the item is just too large for any available box.

Once those are filled, I then evaluate if there are any non-licensed items to be packed. I do multiple passes testing those one by one. "Can this item fit in the existing box". If yes, add it, if no move on to the next item.  I do this step using the VolumePacker directly. 

There are a few additional steps as part of the process which are variants of the above. 

Having given this some extra thought today, I think it would probably only make sense to consider as an improvement if you also considered adding the latest code/comments as part of #169 as a new feature as well. The two go hand in hand.
